pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -375,18 +419,16 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings)
 	switch tls.Mode {
 	case networking.TLSSettings_DISABLE:
 		// TODO: Need to make sure that authN does not override this setting
+		// We remove the TlsContext because it can be written because of configmap.MTLS settings.
+		cluster.TlsContext = nil
 	case networking.TLSSettings_SIMPLE:
 		cluster.TlsContext = &auth.UpstreamTlsContext{
 			CommonTlsContext: &auth.CommonTlsContext{
 				ValidationContext: certValidationContext,
 			},
 			Sni: tls.Sni,
 		}
-		if cluster.Http2ProtocolOptions != nil {
-			// This is HTTP/2 cluster, advertise it with ALPN.
-			cluster.TlsContext.CommonTlsContext.AlpnProtocols = ALPNH2Only
-		}
```

Comment:
why are you removing this?

Reply:
It's correctly done in line 452. Here the TlsContext is still nil.

Reply:
its not. this is for tls simple. not mutual

Reply:
Okay, I get both of your's point.

Because of the incorrect merge, somehow the `Http2ProtocolOptions` is in `ISTIO_MUTUAL`, `MUTUAL`  case, and causing NPR seg fault in pilot. But IIUC, that's should be for `SIMPLE` mode. Is that right, @rshriram ? I can add it back to SIMPLE in that case.

Reply:
you're right. sorry. That code should be there for the SIMPLE case. I guess it was some bad merge